runtime/compiler/control/CompilationThread.cpp

```diff
@@ -8514,7 +8514,7 @@ TR::CompilationInfoPerThreadBase::wrappedCompile(J9PortLibrary *portLib, void *
          if (that->_methodBeingCompiled->isOutOfProcessCompReq())
             {
             OMRProcessorDesc JITClientProcessorDesc = TR::Compiler->target.cpu.getProcessorDescription();
-            target.cpu = TR::CPU(JITClientProcessorDesc);
+            target.cpu = TR::CPU::customize(JITClientProcessorDesc);
```

Comment:
This is not ideal. JITServer should obey exactly what the client tells it to do and not any customization of its own.

Reply:
All customizations are contained in the input parameter `JITClientProcessorDesc`. JITServer is not doing anything extra.`TR::CPU::customize()` is essentially just a wrapped function guarding the constructor. The `_featureMasks` should be the same on JITClient and JITServer. It's a codebase setting that both JITClient and JITServer must be consistent with, therefore `_featureMasks` is not changing anything client wants.

Reply:
The intent of `_featureMasks` is never to modify anything, it's a mask to turn off unnecessary bits so that we can do the compatibility check.

For example:
                 feature             A B C D
underlying feature:      0b1 0 0  1

feature mask:               0b0 0 1 1
(our compiler only uses feature C and D)
after applying mask:    0b0 0 0 1

Wiping off feature A does not equal to disable feature A. A is never used by our compiler so wiping it off has no effect.

Maybe I should change the function names `detect()` and `customize` to just `createCPU()`. They just generate `TR::CPU` object for you based on different input.

Reply:
`target.cpu = TR::CPU::customize(JITClientProcessorDesc);`
means `target.cpu` has been customized to `JITClientProcessorDesc`.

Reply:
ok. Thanks for the explanation. Maybe you should add as a comment:
```
// Customize target.cpu based on the processor description fetched from the client
target.cpu = TR::CPU::customize(JITClientProcessorDesc);
```